air/src/execution_step/execution_context/scalar_variables.rs

```diff
@@ -62,11 +64,32 @@ use std::rc::Rc;
 /// This struct is intended to provide abilities to work with scalars as it was described.
 #[derive(Default)]
 pub(crate) struct Scalars<'i> {
-    // this one is optimized for speed (not for memory), because it's unexpected
-    // that a script could have a lot of inner folds.
-    pub values: HashMap<String, Vec<Option<ValueAggregate>>>,
-    pub iterable_values: HashMap<String, FoldState<'i>>,
-    pub fold_block_id: usize,
+    // TODO: use Rc<String> to avoid copying
+    /// Local values could be considered as a sparse matrix, where a raw corresponds to a value
+    /// name and contains all its variants were set with respect to a depth. This structure
```

Comment:
Please put "value" and "name" on one line, it will be easier to read. Also "contains all its variants were set with respect to a depth" is a bit cryptic, possibly it is simpler to use "column" to describe this part (you can also keep this sentence, just add something about columns). Also "row", not "raw".